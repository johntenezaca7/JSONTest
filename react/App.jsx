import React from "react";
import { Route } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store";

import Landing from "./Landing";
import Test from "./Assessment";
import Done from "./Done";

const App = () => (
  <Provider store={store}>
    <div className="app">
      <Route exact path="/" component={props => <Landing />} />
      <Route exact path="/assessment" component={props => <Test />} />
      <Route exact path="/done" component={props => <Done />} />
    </div>
  </Provider>
);

export default App;
