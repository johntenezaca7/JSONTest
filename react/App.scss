$primary-color1: rgb(255, 255, 255);
$primary-font: 'Gugi', cursive;
$second-font :'Roboto', sans-serif;

* {
    box-sizing: border-box
}
body {
    margin: 0%;
    font-family: $second-font;
    background-color: $primary-color1;
    color: rgb(32, 32, 32);
    overflow:hidden;
  }
  img {
    cursor: pointer;
  }
  
  a {
    color: black;
    text-decoration: none;
    cursor: pointer;
  }
  input {
      padding: 0px;
  }

  video {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      overflow: hidden;
      
  }
  .video-overlay {
      height: 100%;
      width: 100%;
      position: absolute;
      top:0;
      left: 0;
      background-color: rgb(53, 87, 126);
      z-index:1;
      opacity: 0.85;   
      overflow: hidden;
  }

  .landing-video{
      display: gird;
      grid-template-columns: auto;
      justify-content: center;
  }

  .landing {
    background-color: inherit;
    position: absolute;
    top:25%;
    left:25%;
    right:25%;
    margin: auto;
    display: grid;
    grid-template-columns: auto;
    z-index: 1;
    color: white;
    align-self: center;

    h2 {
        font-family: $primary-font; 
        font-size: 50px;
        margin: 0%;
        justify-self: center;
    }

    .GetName {
        text-align: center;
        margin-bottom: 2%;
        font-family: $primary-font;
        justify-self: center
    }

    div {
    justify-self: center;
    }
}
  
  .inside-landing input {
    height: 40px;
    font-size: 20px;
    text-align: center;
  }
  .landing-btn {
    margin-top: 5%;
    background-color: rgb(255, 255, 255); 
    border: none;
    width: 100%;
    color: white;
    padding: 0px;
    height: 40px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;
    justify-self: center
  }
  
  .landing-btn {
    background-color: white;
    color: black;
  }
  
  .landing-btn:hover {
    background-color: rgb(35, 55, 78);
    color: white;
  }
  
  .test-display h2 {
    text-align: center;
    margin-bottom: 5%;
  }
  
  .test-display {
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: 1fr 4fr
  }
  
  .test-box {
    height: 500px;
    width: 900px;
    border: 1px solid rgb(250, 250, 250);
    background-color: white;
    border-radius: 2%;
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    justify-self: center;
  }
  
  .test-details{
    text-align: center;
  }
  .test-details h4 {
    font-size: 25px;
    margin: 1%;
  }
  .test-details p {
    font-size: 16px;
    line-height: 1.5;
  }
  .container {
    padding: 5%;
  }
  .back-n-count {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }
  
  .back-n-count p {
    justify-self: end;
  }
  
  .answers {
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box {
    border: 1px solid grey;
    width: 180px;
    height: 100px;
    border-radius: 15px;  
    box-shadow: -2px 3px 5px rgba(77, 78, 79, 0.2);
    margin: 1%;
    cursor: pointer;
    display: flex;
    flex-flow: row wrap;
    justify-content: center
  }
  
  .answer-box p {
    align-self: center;
    padding-left: 5px;
  
  }