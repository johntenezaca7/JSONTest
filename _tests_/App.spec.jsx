import React from "react";
import jest from "jest";
import { shallow } from "enzyme";
import App from "../react/App";

test("App Component renders correctly", () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
