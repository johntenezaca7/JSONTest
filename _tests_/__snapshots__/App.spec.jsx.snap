// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/assessment"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/done"
    />
  </div>
</Provider>
`;
